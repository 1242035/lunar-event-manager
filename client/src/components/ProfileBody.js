import React from 'react';
import ProfileUserCard from './ProfileUserCard';
import DiscoverTitleCard from '../components/DiscoverTitleCard';
import ConnectedProfileEventCards from './ProfileEventCards';
/**
 * Houses contnet of the profile page.
 * @param{undefined}
 * @return{React.Component}
 */
const ProfileBody = props => (
  <div className="io-body-content">
    <div className="container-fluid">
      <ProfileUserCard />
      <DiscoverTitleCard>Your Events</DiscoverTitleCard>
      <ConnectedProfileEventCards
        showModifyEventModal={props.showModifyEventModal}
        showAlertModal={props.showAlertModal}
        showLoader={props.showLoader}
        hideLoader={props.hideLoader}
        showNotification={props.showNotification}
        showConfirmModal={props.showConfirmModal}
      />
    </div>
  </div>
);

export default ProfileBody;
