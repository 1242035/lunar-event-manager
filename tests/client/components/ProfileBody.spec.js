import React from 'react';
import { shallow } from 'enzyme';
import ProfileBody from '../../../client/src/components/ProfileBody';

describe('ProfileBody', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<ProfileBody />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = shallow(<ProfileBody />);
    expect(wrapper.length).toEqual(1);
  });
});
