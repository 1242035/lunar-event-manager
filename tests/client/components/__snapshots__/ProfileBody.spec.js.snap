// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileBody Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileBody />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container-fluid"
      >
        <Connect(ProfileUserCard) />
        <DiscoverTitleCard>
          Your Events
        </DiscoverTitleCard>
        <Connect(ProfileEventCards) />
      </div>,
      "className": "io-body-content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ProfileUserCard) />,
          <DiscoverTitleCard>
            Your Events
          </DiscoverTitleCard>,
          <Connect(ProfileEventCards) />,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Your Events",
          },
          "ref": null,
          "rendered": "Your Events",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hideLoader": undefined,
            "showAlertModal": undefined,
            "showConfirmModal": undefined,
            "showLoader": undefined,
            "showModifyEventModal": undefined,
            "showNotification": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container-fluid"
        >
          <Connect(ProfileUserCard) />
          <DiscoverTitleCard>
            Your Events
          </DiscoverTitleCard>
          <Connect(ProfileEventCards) />
        </div>,
        "className": "io-body-content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(ProfileUserCard) />,
            <DiscoverTitleCard>
              Your Events
            </DiscoverTitleCard>,
            <Connect(ProfileEventCards) />,
          ],
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Your Events",
            },
            "ref": null,
            "rendered": "Your Events",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "hideLoader": undefined,
              "showAlertModal": undefined,
              "showConfirmModal": undefined,
              "showLoader": undefined,
              "showModifyEventModal": undefined,
              "showNotification": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
